src/App.tsx

```diff
@@ -1,14 +1,38 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+
+import { getFilteredTodos } from './utils/getFilteredTodos';
+
+type CurrentTodo = Todo | null;
+
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<FilterType>(FilterType.All);
+  const [query, setQuery] = useState('');
+  const [todo, setTodo] = useState<CurrentTodo>(null);
+
+  const filteredTodos = getFilteredTodos(todos, filterBy, query);
+
+  useEffect(() => {
```

Comment:
Think about using custom hook for this

Reply:
For the current case it is ok to keep all as it is, but if you had to reload Todos from server on each filter change, it would make total sense to combine effect hook + Todos filtering in the custom hook (like `useTodos`)